packages/webapp/src/main/components/apollon-editor-component/ApollonEditorComponent.tsx

```diff
@@ -46,6 +68,42 @@ export const ApollonEditorComponent: React.FC = () => {
     initializeEditor();
   }, [containerRef.current, createNewEditor]);
 
+  useEffect(() => {
+    const previewDiagram = async () => {
+      if (
```

Comment:
I suggest combining this into a single useEffect to improve readability and reduce code duplication. Having two useEffect hooks here makes the logic harder to follow